lib/luadevice.c

```diff
@@ -240,6 +279,40 @@ static int luadevice_stop(lua_State *L)
 	return 0;
 }
 
+/***
+* Creates and installs a new character device driver in the system.
+* This function binds a Lua table (the `driver` table) to a new
+* character device file (`/dev/<name>`), allowing Lua functions
+* to handle file operations on that device.
+*
+* @function new
+* @tparam table driver A table defining the device driver's properties and callbacks.
+*   It **must** contain the field:
+*   @field name string The name of the device. This name will be used to create the device file `/dev/<name>`.
+*   It **might** optionally contain the following fields (callback functions):
+*   @field open function Callback for the `open(2)` system call. Signature: `function(driver_table)`. Expected to return nothing.
+*   @field read function Callback for the `read(2)` system call. Signature: `function(driver_table, length, offset) -> string [, updated_offset]`. Receives the driver table, the requested read length (integer), and the current file offset (integer). Should return the data as a string and optionally the updated file offset (integer). If `updated_offset` is not returned, the offset is advanced by the length of the returned string (or the requested length if the string is longer).
+*   @field write function Callback for the `write(2)` system call. Signature: `function(driver_table, buffer_string, offset) -> [written_length] [, updated_offset]`. Receives the driver table, the data to write as a string, and the current file offset (integer). May return the number of bytes successfully written (integer) and optionally the updated file offset (integer). If `written_length` is not returned, it's assumed all provided data was written. If `updated_offset` is not returned, the offset is advanced by the `written_length`.
+*   @field release function Callback for the `release(2)` system call (called when the last file descriptor is closed). Signature: `function(driver_table)`. Expected to return nothing.
+*   @field mode integer Optional file mode flags (e.g., permissions) for the device file. Use constants from the `linux.stat` table (e.g., `linux.stat.IRUGO`).
+* @treturn userdata A Lunatik object representing the newly created device. This object can be used to explicitly stop the device using the `:stop()` method.
+* @raise Error if the device cannot be allocated or registered in the kernel, or if the `name` field is missing or not a string.
```

Comment:
can you break lines here? I'm not sure how many columns here.. but it looks very long..